lib/core/src/display/symbol.rs

```diff
@@ -216,22 +238,26 @@ impl Symbol {
     /// Creates a new VertexArrayObject, discovers all variable bindings from shader to geometry,
     /// and initializes the VAO with the bindings.
     fn init_variable_bindings(&mut self, var_bindings:&[shader::VarBinding]) {
-        self.vao      = Some(VertexArrayObject::new(&self.context));
-        self.uniforms = default();
-        self.textures = default();
+        const MAX_TEXTURE_UNITS:u32 = 16;
```

Comment:
Dont hardcode 16 slots - check it!
Sample API usage: https://webglreport.com/?v=2
Stats: https://webglstats.com/webgl/parameter/MAX_TEXTURE_IMAGE_UNITS
As you can see, many devices nowadays do not support 16 textures. 4% of devices support 32 textures.

Reply:
Ok, I read the documentation of OpenGl, where 16 is a minimum. And I can read MAX_TEXTURE_IMAGE_UNITS, but the question is if we want to make library which may behave differently on user machines?